### PR TITLE
Warning if on wrong account for some tabs

### PR DESCRIPTION
**Pull request by zjmiller**
Big diff for second file is just wrapping everything in <></> and adding <AccountWarning> component at top

### CODE REVIEW

File: components/AccountWarning.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  const inRelevantTab =
+    router.query.tab &&
+    typeof router.query.tab === 'string' &&
+    ['upcoming_payments', 'manage_listings', 'sales'].includes(router.query.tab)
+
+  if (!inRelevantTab) {
+    return null
+  }
+
+  const doesUrlAddressEqualAccountAddress = id === accountData?.address
```
Comment (SentientClee) [inline components/AccountWarning.tsx:18]: Do we need to worry about casing?

File: components/AccountWarning.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      ? 'to manage your listings'
+      : router.query.tab === 'sales'
+      ? 'to view your past sales'
+      : 'to access this page'
+
+  // If no connected account at all
+  if (id !== accountData?.address && !accountData?.address) {
+    return (
+      <div className="flex w-screen items-center justify-center gap-2 bg-[#FFA500] p-4 text-black">
+        You must connect a wallet to {tabDependentMsgPart}
```
Comment (SentientClee) [inline components/AccountWarning.tsx:40]: Double `to`

Comment (SentientClee) [reply components/AccountWarning.tsx:40]: "to to"

File: components/AccountWarning.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      </div>
+    )
+  }
+
+  // If there is a connected account, but it doesn't match
+  if (id !== accountData?.address && accountData?.address) {
+    return (
+      <div className="flex w-screen items-center justify-center gap-2 bg-[#FFA500] p-4 text-black">
+        You are connected with {truncateAddress(accountData.address)}, but need
+        to be connected with {truncateAddress(id)} to access this page
```
Comment (SentientClee) [inline components/AccountWarning.tsx:50]: Do you want the dependent message type here as well? Works better without it I suppose

**Review by SentientClee — APPROVED** [review]

File: components/AccountWarning.tsx (hunk 2 above)
Comment (zjmiller) [reply components/AccountWarning.tsx:40]: nice catch

**Pull request merged**